crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -444,23 +457,25 @@ impl<'w, 's> Commands<'w, 's> {
     #[inline]
     #[track_caller]
     pub fn entity(&mut self, entity: Entity) -> EntityCommands {
-        #[inline(never)]
-        #[cold]
-        #[track_caller]
-        fn panic_no_entity(entities: &Entities, entity: Entity) -> ! {
-            panic!(
-                "Attempting to create an EntityCommands for entity {entity}, which {}",
-                entities.entity_does_not_exist_error_details_message(entity)
-            );
+        if !self.entities.contains(entity) {
+            match self.error_mode.unwrap_or(CommandErrorMode::Panic) {
+                CommandErrorMode::Silent => (),
+                CommandErrorMode::Warning => warn!(
+                    "Attempted to create an EntityCommands for entity {entity}, which does not exist; returned invalid EntityCommands"
+                ),
+                CommandErrorMode::Error => error!(
+                    "Attempted to create an EntityCommands for entity {entity}, which does not exist; returned invalid EntityCommands"
+                ),
+                CommandErrorMode::Panic | CommandErrorMode::Custom(_) => panic!(
```

Comment:
Aha, here's the panic. This should be called out more clearly at least, but I think that passing down a more complex Result or adding an error arm is the better way. The custom error handler should be able to know whether or not the entity failed to exist at all, and respond accordingly.

Reply:
I was kind of hoping we wouldn't have to put handling logic on this level. I feel like this sort of match statement should ideally only appear once, and everything should be funneled through it on a lower level. That's just a feeling on my part.

Reply:
Do we need to remove the panic here? I feel like if the user doesn't want it to panic they should just be using `get_entity`.

Reply:
Eventually I think the plan is to collapse our panicking and non-panicking variants into the same call.

Reply:
Yeah, I'd like to remove `get_entity` completely in favor of this error handling technique. It's frustrating that there's so many different edge cases to guard for, and no unified way to make sure you've handled them all.

Reply:
That makes sense, but I still feel like this pr should be focused on the deferred errors. Or are you saying that creating a EntityCommands will only panic when the command queue is applied?

Reply:
> That makes sense, but I still feel like this pr should be focused on the deferred errors

I agree with that, but I think we can do some future-proofing here. It's fine to leave it for follow-up though: I just think it's a bit messy as is.